### PR TITLE
feat(volcengine-ast2): Chinese↔English bidirectional mode (closes #229)

### LINKED ISSUE #229: feat(volcengine_ast2): expose Chinese↔English bidirectional (zhen) mode in language selectors
## Context

User feedback from a classroom use case (Youth Night School English learning group): when picking **Doubao AST 2.0** as the provider, there is no UI path to fully enable Doubao's auto Chinese↔English bidirectional mode. Doubao AST 2.0 supports this via:

```
source_language = "zhen"
target_language = "zhen"
```

The desired behavior is:

- When the user speaks Chinese → Sokuji outputs English.
- When the user speaks English → Sokuji outputs Chinese.

No need to know in advance which language the speaker will use next.

## Current state

Reviewing the code, the underlying `VolcengineAST2Client` already forwards `sourceLanguage`/`targetLanguage` into the WebSocket `StartSession` request, so the realtime path is already capable of running in `zhen`/`zhen` mode. The gap is in the provider config and UI exposure:

`src/services/providers/VolcengineAST2ProviderConfig.ts`:

```ts
static getSourceLanguages(): LanguageOption[] {
  return VolcengineAST2ProviderConfig.BIDIRECTIONAL_LANGUAGES; // includes 'zhen'
}

static getTargetLanguages(): LanguageOption[] {
  return VolcengineAST2ProviderConfig.LANGUAGES; // does NOT include 'zhen'
}
```

So `zhen` is reachable as a source language but not as a target language. The Simple UI Language section also doesn't pair the two sides together when the user picks the bidirectional option, which means the effective state ends up like `auto → en` instead of the intended `zhen → zhen`.

## Proposed change

1. Make `zhen` selectable on **both** source and target language lists for the `volcengine_ast2` provider (e.g. have `getTargetLanguages()` return `BIDIRECTIONAL_LANGUAGES` as well, or expose a dedicated bidirectional option).
2. In `LanguageSection.tsx` (Simple mode), when the user selects the Chinese↔English bidirectional option on either side, synchronize the other side so both `sourceLanguage` and `targetLanguage` are written as `'zhen'`.
3. In `ProviderSpecificSettings.tsx` (Advanced mode), make sure the provider-specific selectors also offer `zhen` on both sides and keep them in sync when one side is changed to the bidirectional value.
4. Label the option clearly, e.g. "Chinese-English Bidirectional" / "中英双语 (zh↔en)", consistent with the label already defined in the provider config.

## Acceptance criteria

- For provider Doubao AST 2.0, the user can pick a single "Chinese-English Bidirectional" option that results in `sourceLanguage === 'zhen' && targetLanguage === 'zhen'`.
- Switching to this option from any other source/target combination updates both sides atomically (no transient `auto → en` style state).
- A live session started in this mode auto-translates zh→en when the speaker uses Chinese and en→zh when the speaker uses English, with no further config.

## Reporter

Reported by **Ray Chow**, who is running Sokuji in a Youth Night School English learning group and diagnosed the gap down to the code level — credit to him for the detailed pointer into `VolcengineAST2ProviderConfig.ts`, `LanguageSection.tsx`, and `ProviderSpecificSettings.tsx`.

### PR DESCRIPTION
**Pull request by jiangzhuo**
## Summary

Exposes Doubao AST 2.0's Chinese↔English bidirectional mode (`source_language = target_language = 'zhen'`) through the existing source/target language dropdowns. Picking the bidirectional entry on either side **atomically syncs the other side**, so the settings never sit in a server-invalid `zhen/<other>` state.

When this mode is active, the server outputs English when the user speaks Chinese, and Chinese when the user speaks English — no further configuration required.

Reported by Ray Chow (Youth Night School English learning group). Closes #229.

## Changes

| Commit | Layer |
|---|---|
| `60a0542f` | docs(specs): design document |
| `d92209df` | docs(plans): implementation plan |
| `0d87cecc` | feat: expose `'zhen'` on target-language list (Advanced mode static method) |
| `017a4edc` | feat: add pure helper `resolveAST2LanguagePair` + 9 unit tests |
| `580aec7e` | feat: atomic `zhen` sync in `LanguageSection` (Simple mode) |
| `450e70a5` | feat: disable swap button in `zhen/zhen` state |
| `4bb78d9b` | feat: atomic `zhen` sync in `ProviderSpecificSettings` (Advanced mode) |
| `e6e47a27` | fix: expose `'zhen'` in Simple-mode list via `getConfig().languages` |
| `2ee2e4cc` | fix: swap button no longer clobbers source with stale closure read |

**Files touched:** `VolcengineAST2ProviderConfig.ts`, `LanguageSection.tsx`, `ProviderSpecificSettings.tsx`, plus new `volcengineAST2LanguageSync.ts` and its test.

## How it works

The helper `resolveAST2LanguagePair(current, change)` is a pure transform encoding 4 rules from the spec ([§2](docs/superpowers/specs/2026-05-14-volcengine-ast2-bidirectional-mode-design.md)):

- R1/R2 — picking `'zhen'` on either side forces both fields to `'zhen'`.
- R3/R4 — leaving `'zhen'` on one side (when state was `zhen/zhen`) resets the other side to its provider default (`zh` source / `en` target).
- Passthrough — ordinary single-side edits don't touch the other side.

Both UI surfaces (`LanguageSection` Simple mode + `ProviderSpecificSettings` Advanced mode) call this helper and write the result in a single `updateVolcengineAST2Settings(...)` action.

## Non-goals

- **`ja → de` style invalid combos** — AST2 also requires one of source/target to be `zh` or `en` in non-bidirectional mode. Pre-existing gap; out of scope for this PR.
- **i18n of the `'zhen'` label** — the existing `中英双语 (zh↔en)` is bilingual already.
- **Generalised bidirectional architecture** — AST2 exposes exactly one bidirectional value; designed as a single sentinel.

## Test plan

- [x] 9 new Vitest unit tests covering R1, R1', R2, R3, R3-degenerate, R4, R4-degenerate, 2 passthrough cases — all green.
- [x] Full test suite: 27 files / 302 tests pass.
- [x] TypeScript: no new errors in touched files (4 pre-existing TS6133 unused-import warnings unaffected).
- [x] Manual smoke (Electron, Simple mode):
  - source → `中英双语` ⇒ target auto-syncs ✓
  - target → `中英双语` ⇒ source auto-syncs ✓
  - `zhen/zhen` → change source to `ja` ⇒ `ja/en` ✓
  - `zhen/zhen` → change target to `fr` ⇒ `zh/fr` ✓
  - `zhen/zhen` → swap button is disabled ✓
- [x] Manual smoke (Electron, Advanced mode): same four dropdown-sync checks ✓
- [x] Swap-button regression check (`zh/ja` → swap → `ja/zh`) ✓ — caught a post-handoff bug, fixed in `2ee2e4cc`.
- [ ] End-to-end live session with `zhen/zhen`: confirm zh→en and en→zh switch within one session.

## Known follow-ups (not in this PR)

- The `from_language` analytics field is present on Advanced-mode `language_changed` events but absent on Simple-mode trailing emits (pre-existing asymmetry, preserved by both call sites).
- The "non-bidirectional needs `zh` or `en` on at least one side" server constraint isn't surfaced in UI.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Enabled Volcengine AST2 Chinese↔English bidirectional (zhen) mode
  * Automatic two-way synchronization of source/target languages, with swap-button disabled for the zhen pair
  * Provider language set aligned to bidirectional mode

* **Tests**
  * Added unit tests covering language-pair sync, resets, and edge cases

* **Documentation**
  * New rollout and design docs plus a manual smoke runbook and self-review checklist

[Review Change Stack](https://app.coderabbit.ai/change-stack/kizuna-ai-lab/sokuji/pull/235)

### CODE REVIEW

File: src/components/Settings/sections/LanguageSection.tsx [hunk 1]
```diff
@@ -153,9 +154,27 @@ const LanguageSection: React.FC<LanguageSectionProps> = ({
...
+        // Spec §5: when bidirectional sync also changes the other side, emit a
+        // second analytics event so consumers see both transitions. The trailing
+        // `trackEvent` below this switch still emits the source-side event.
+        if (next.targetLanguage !== prev.targetLanguage) {
+          trackEvent('language_changed', {
+            to_language: next.targetLanguage,
+            language_type: 'target',
+          });
+        }
         break;
```
[the remark this answers is not in the document]
Comment (jiangzhuo) [reply src/components/Settings/sections/LanguageSection.tsx]: Adopted in 6470ce1e. Both source-side and target-side AST2 branches now emit source→target inside the case body and `return` to skip the trailing emit. Ordering matches the spec §5 example and the Advanced-mode call sites.

File: src/services/providers/volcengineAST2LanguageSync.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    return { sourceLanguage: ZHEN, targetLanguage: ZHEN };
+  }
+
+  // R3 / R4: leaving 'zhen' on one side while the other was also 'zhen' means
+  // we must reset the other side to its provider default to avoid a transient
+  // 'zhen / <other>' state that the server rejects.
+  if (change.side === 'source' && current.sourceLanguage === ZHEN && current.targetLanguage === ZHEN) {
+    return { sourceLanguage: change.value, targetLanguage: DEFAULT_TARGET };
+  }
+  if (change.side === 'target' && current.sourceLanguage === ZHEN && current.targetLanguage === ZHEN) {
```
[the remark this answers is not in the document]
Comment (jiangzhuo) [reply src/services/providers/volcengineAST2LanguageSync.ts]: Adopted in de9f93a4. Generalized R3/R4 into a single rule: whenever the user picks a non-'zhen' value AND the OTHER side is currently 'zhen', reset the other side to its server-contract default. Subsumes the original zhen/zhen → non-zhen case and additionally cleans up legacy zhen/<other> persisted pairs from older builds (e.g. zhen/en + target=fr now correctly produces zh/fr instead of zhen/fr).

File: src/services/providers/volcengineAST2LanguageSync.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+export interface AST2LanguageChange {
+  side: 'source' | 'target';
+  value: string;
+}
+
+const ZHEN = 'zhen';
+const DEFAULT_SOURCE = 'zh';
+const DEFAULT_TARGET = 'en';
```
[the remark this answers is not in the document]
Comment (jiangzhuo) [reply src/services/providers/volcengineAST2LanguageSync.ts:36]: Pushing back on this one. The literals are anchored to the **server contract** (AST 2.0 rejects non-bidirectional pairs unless one of source/target is 'zh' or 'en'), not to the provider's UI defaults. The helper's reset target MUST be in {'zh', 'en'} for the resulting pair to be server-valid — if the UI default ever changed to, say, 'ja/en', sourcing the helper's reset from `VolcengineAST2ProviderConfig.defaults` would silently emit an invalid pair to the server. Added a comment in de9f93a4 making this explicit.

File: src/services/providers/volcengineAST2LanguageSync.test.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        { sourceLanguage: 'zhen', targetLanguage: 'zhen' },
+        { side: 'target', value: 'zh' },
+      ),
+    ).toEqual({ sourceLanguage: 'zh', targetLanguage: 'zh' });
+  });
+
+  // Normal (non-bidirectional → non-bidirectional) update: passes through
+  // without touching the other side. Not in the spec's rule table, but the
+  // helper must support being called for every change to be useful as the
+  // single VOLCENGINE_AST2 update path in the UI.
```
[the remark this answers is not in the document]
Comment (jiangzhuo) [reply src/services/providers/volcengineAST2LanguageSync.test.ts:77]: Adopted in de9f93a4. Added 4 tests covering legacy zhen/en and en/zhen permutations on both source-side and target-side changes. Helper test count is now 13 (9 original + 4 legacy).

File: docs/superpowers/specs/2026-05-14-volcengine-ast2-bidirectional-mode-design.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+### 5. Analytics
+
+Existing `language_changed` `trackEvent` calls continue to fire — but in R1/R2/R3/R4 the helper updates two fields. Emit **two** `language_changed` events (one for `source`, one for `target`) with the post-state value so analytics consumers see a coherent picture: e.g. picking `zhen` on the source fires `{to_language: 'zhen', language_type: 'source'}` followed by `{to_language: 'zhen', language_type: 'target'}`.
+
+This adds noise (two events per user action when only one click happened) but matches the actual settings change. Alternative — emitting a single `bidirectional_mode_enabled` event — would require schema additions. Two events is cheaper.
+
+### 6. Testing
+
+**Unit (Vitest)**: a single test file covering the helper applies R1–R4 to a mocked store and asserts the post-state plus that `updateVolcengineAST2Settings` was called **exactly once** per user action.
```
[the remark this answers is not in the document]
Comment (jiangzhuo) [reply docs/superpowers/specs/2026-05-14-volcengine-ast2-bidirectional-mode-design.md]: Adopted in 52cd8de6. Spec §6 now describes the actual pure-function test approach; the 'exactly once' guarantee is preserved at the call site (each writes both fields in one `updateVolcengineAST2Settings` call) and verified by code reading, not by unit tests on the helper itself.

File: src/services/providers/volcengineAST2LanguageSync.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jiangzhuo) [reply src/services/providers/volcengineAST2LanguageSync.ts:36]: Same response as the duplicate Copilot comment above: the constants are anchored to the AST 2.0 server contract (one of source/target must be 'zh' or 'en' in non-bidirectional mode), not to the provider's UI defaults. Sourcing from `VolcengineAST2ProviderConfig.defaults` would silently emit an invalid pair if the UI defaults ever legitimately diverged. Clarifying comment added in de9f93a4.
